renderdoc/driver/metal/metal_device.cpp

```diff
@@ -430,6 +480,20 @@ bool WrappedMTLDevice::Serialise_newTextureWithDescriptor(SerialiserType &ser,
 
   if(IsReplayingAndReading())
   {
+    // Ensure the created textures can be read by a shader
+    // Metal driver will treat TextureUsageUnknown as all options
+    if(descriptor.usage != MTL::TextureUsageUnknown)
+      descriptor.usage = (MTL::TextureUsage)(descriptor.usage | MTL::TextureUsageShaderRead);
```

Comment:
Just to check - is there any possible side-effect of making this change in metal, or is it entirely independent?

On D3D11 e.g. we could freely set these bits only on replay and it wouldn't change anything. On Vulkan at least we need to set these bits during capture too to ensure that queried memory properties are respected, as they can change with different usage flags. I assume metal falls into the former camp, but it's worth making sure.